### ISSUE TITLE
We're updating the updated_at field in metabase_table on every field even though there's no change

### ISSUE THREAD

**Issue by paoliniluis** [metabase/metabase#44946]
### Describe the bug

We started updating the updated_at field on the metabase_table table on recent versions although there's no change in the table. This lead to a problem on a customer who was using a trigger on that field, which started firing on every single sync no matter if there was a change or not

### To Reproduce

1) start v47 and add a table to the DW
2) sync it, see the updated_at field not changing
3) now move to v48 or v49, do the syncs, see that the updated_at field having updated values on every sync

### Expected behavior

We should not change stuff on the DB if there's no change

### Logs

NA

### Information about your Metabase installation

```JSON
v48+
```

### Severity

P3

**Comment by luizarakaki** [comment]
@calherries Is this because of the `view_count` column?
I think this is more severe than P3 if `updated_at` is refreshing on every read.

**Comment by bshepherdson** [comment]
No, actually, this isn't related to `view_count`.

**Comment by bshepherdson** [comment]
This one starts out as an update of `result_metadata` on the card after we execute its query.

The flow is:
1. start a transaction
2. `SELECT` the card
3. compare its values with the new ones, and drop those that are `=` from the update
4. issue the `UPDATE` for the things that actually changed

which would be fine except that we already added `:updated_at (now)` to the updates. So the only real update (to `result_metadata`) got removed, and we don't have a special case to drop an `UPDATE` like that.

**Comment by calherries** [comment]
@paoliniluis is this a postgres database? I suspect this is because of https://github.com/metabase/metabase/pull/39494

**Comment by calherries** [comment]
@paoliniluis do you know? `updated_at` should not change unless the estimated number of rows in the table changes, from `pg_stat_user_tables.n_live_tup`. in that case we'll update `metabase_table.estimated_row_count` and also `updated_at`.

**Comment by paoliniluis** [comment]
@calherries nope, in fact I was doing this test on the sample database, with no change in rows at all

**Comment by calherries** [comment]
Okay thanks for clarifying, that's a separate issue then. Even if we fix that, because of `estimated_row_count`, do you think that `metabase_table.updated_at` should not change even if `estimated_row_count` changes? Because if a customer is using a Postgres DW, a trigger on `updated_at` will fire much more frequently than before https://github.com/metabase/metabase/pull/39494

**Comment by luizarakaki** [comment]
I think it is fine to update `updated_at` if `estimated_row_count` changes... I'm not 100% confident, but doesn't look terrible because there is indeed a change in the entity.

But analytical properties like view_count shouldn't update updated_at because the entity isn't changing
